nexus/types/src/deployment.rs

```diff
@@ -307,9 +307,97 @@ impl Blueprint {
         }
     }
 
+    /// Iterate over the in-service [`BlueprintZoneConfig`] instances in the
+    /// blueprint, along with the associated sled id.
+    pub fn in_service_zones(
+        &self,
+    ) -> impl Iterator<Item = (SledUuid, &BlueprintZoneConfig)> {
+        // Danger note: this call has no danger of accessing expunged zones,
+        // because we're filtering to in-service.
+        self.danger_all_omicron_zones(BlueprintZoneDisposition::is_in_service)
```

Comment:
I'd consider removing `danger_all_omicron_zones` and just inlining it into the callers since it's not that complicated.